### ISSUE TITLE
_nodeUtil.js throws caught exception

### ISSUE THREAD

**Issue by krzysztofp** [lodash/lodash#2920]
I'm debugging React Native application and this part keeps on throwing the caught exception, which slows the debugging process:
![image](https://cloud.githubusercontent.com/assets/1634660/21753312/09358362-d5eb-11e6-95bc-19b1a39b9993.png)

Can you add a simple checking if the object and it's property is defined instead of a dummy catch?

**Comment by jdalton** [comment]
Hi @krzysztofp!

There are several guards in place
```js
freeProcess && freeProcess.binding && freeProcess.binding('util')
```
I'm not sure what else to do. You can always limit your debugging to uncaught exceptions or use the just-my-code features of them to ignore third party libraries.

**Comment by krzysztofp** [comment]
@jdalton - you shouldn't been using try-catch for that, use for example:

    var nodeUtil = (function() {
        if (freeProcess && freeProcess.binding) {
            try {
                return freeProcess.binding('util');
            } catch(e) {}
        }
    }());

**Comment by jdalton** [comment]
In my snippet if `freeProcess` or `freeProcess.binding` is `undefined` no error will be thrown which is why it's a bit confusing. I made the snippet the way it is to avoid noisey debug and the users confirmed the fix so I'm left scratching my head on this one. My code is valid and won't throw if `binding` is `undefined`. Moving the try-catch won't effect anything here.

**Comment by krzysztofp** [comment]
Ok now it's more clear to me:
![image](https://cloud.githubusercontent.com/assets/1634660/21776675/e81665dc-d69b-11e6-97ec-9c4e301fd6e0.png)

It looks this commit was implemented in 4.17.0 while I'm using 4.16.4 version. So I hope when I update to ^4.17.0 all should be good.

It also looks React Native it using the 3.10.1 version, so I'll need to wait for them to update as well:
![image](https://cloud.githubusercontent.com/assets/1634660/21776714/0d86c8de-d69c-11e6-8710-3fcb1e36974e.png)


But becuase of `[this commit](https://github.com/lodash/lodash/commit/6d951ccc87f3cc867252a507e7d177c12badf3d5)`
I think this ticket can be closed.
